### ISSUE TITLE
Enable detecting "See buying options" on Amazon

### ISSUE THREAD

**Issue by sarmadym** [samuelm2/Nvidia-Notify#15]
Amazon only shows see buying options on the 3080/3090 page and not directly add to card therefore the bot doesn't detect it. Can this be addressed? I saved the html of the Amazon page when the see buying options showed can provide if it helps. I changed the detection string to "See buying options" in the code but then it detects everything. Need something more specific to the button when it is actually showing on the page not source of html

![image](https://user-images.githubusercontent.com/3420786/94375967-63fd7d80-00e5-11eb-8cc3-9ca3f6fdceed.png)

**Comment by rambosbff** [comment]
Yeah, I saw this pop up In another discord I'm in, but this one flew right past snatcher

**Comment by jxjftw** [comment]
One of the guys in my discord server got a positive hit, while most others did not. Amazon must not be displaying "see buying options" 100% of the time. 
![image](https://user-images.githubusercontent.com/20994780/94376181-f6eae780-00e6-11eb-97c4-ea2779508ea8.png)

**Comment by samuelm2** [comment]
I was under the impression that the "See buying options" was when it was a 3rd party seller, but the "Add to cart" was for an actual seller. Is that not the case?

**Comment by sarmadym** [comment]
> One of the guys in my discord server got a positive hit, while most others did not. Amazon must not be displaying "see buying options" 100% of the time.
> ![image](https://user-images.githubusercontent.com/20994780/94376181-f6eae780-00e6-11eb-97c4-ea2779508ea8.png)

Which line of the code caught it? This line is not going to catch the screenshot above. I have a local clone of the code which I run and tested it. `"https://www.amazon.com/stores/page/6B204EA4-AAAC-4776-82B1-D7C3BD9DDC82?ingress=0": (">Add to Cart<", True, GET_URLLIB, 'Amazon 3080')`

**Comment by jxjftw** [comment]
The only thing amazon.ts is looking for is add to cart, it must have shown add to cart instead of see all buying options for him. Also last week (wed i think) when the EVGA's dropped on amazon the page was displaying "see all buying options" instead of add to cart as well.

**Comment by jxjftw** [comment]
> I was under the impression that the "See buying options" was when it was a 3rd party seller, but the "Add to cart" was for an actual seller. Is that not the case?

When these pop up they show see buying options instead of add to cart, by the time the page switches to add to cart it's too late.

**Comment by sarmadym** [comment]
> I was under the impression that the "See buying options" was when it was a 3rd party seller, but the "Add to cart" was for an actual seller. Is that not the case?

> The only thing amazon.ts is looking for is add to cart, it must have shown add to cart instead of see all buying options for him. Also last week (wed i think) when the EVGA's dropped on amazon the page was displaying "see all buying options" instead of add to cart as well.

Yes, this time it was "ships and sold by Amazon.com" when you clicked on "See buying options" but still not "Add to cart" shown on the page the bot crawls. So its not 3rd party seller.

**Comment by Kosyne** [comment]
Is there a way to differentiate 3rd party sellers if a method like this was implemented? We wouldn't want to notify if that were the case.

**Comment by jxjftw** [comment]
> Is there a way to differentiate 3rd party sellers if a method like this was implemented? We wouldn't want to notify if that were the case.

you could search for text sold by amazon.com

**Comment by jeckerman** [comment]
- Grab the ASIN from the url in the "See buying options" button
- https://www.amazon.com/ASUS-GeForce-Overclocked-Graphics-ROG-STRIX-RTX-2080-O11G/dp/B07HY6QWXN?ref_=ast_sto_dp
- Reconstruct the url as such:
- https://www.amazon.com/gp/offer-listing/B07HY6QWXN/ref=&f_new=true&f_primeEligible=true

Source: https://www.scrapehero.com/tutorial-how-to-scrape-amazon-seller-prices-using-python/
